src/Components/test/E2ETest/Infrastructure/ServerFixtures/ToggleExecutionModeServerFixture.cs

```diff
@@ -32,7 +35,11 @@ protected override string StartAndGetRootUri()
                 var underlying = new DevHostServerFixture<TClientProgram>();
                 underlying.PathBase = PathBase;
                 _serverToDispose = underlying;
-                return underlying.RootUri.AbsoluteUri;
+                var uri = underlying.RootUri.AbsoluteUri;
+
+                Host = underlying.Host;
+
+                return uri;
```

Comment:
I don't follow why we need to separate the `var uri` and `return uri` here. Couldn't this be the following?

```cs
                Host = underlying.Host;
                return underlying.RootUri.AbsoluteUri;
```

Not a big deal, of course!

Reply:
Accessing the RootUri is required for the side-effects (it starts the server)

I'm not planning to change that in this PR :)

Reply:
Oh I see. It's non-obvious, so could you do a comment like the following?

```cs
var uri = underlying.RootUri.AbsoluteUri; // As a side-effect, this starts the server
```

Longer term we should fix this, as it's super weird (no doubt my fault).

Reply:
Added a comment